app.js

```diff
@@ -0,0 +1,119 @@
+var createError = require('http-errors');
+var express = require('express');
+var app = express();
+const request = require('request');
+const zlib = require('zlib');
+const util = require('util');
+// const websiteUrlArr = ['https://www.zara.com/', 'https://www.sephora.fr/', 'https://www.stories.com/', 'https://www.nike.com/', 'https://www.adidas.com/', 'https://www.saksfifthavenue.com/', 'https://www.uniqlo.com/', 'https://www.cremedelamer.com/', 'http://www.calvinklein.com/'];
+const websiteUrlArr = ['https://www.zara.com/'];
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+
+app.get('/', (req, res) => {
+  res.json('Welcome to the express app')
+});
+
+app.get('/site-urls', async (req, res, next) => {
```

Comment:
should take base url as a parameter instead of hardcoded `websiteUrlArr` array

Reply:
bug?: `GET http://localhost:3000/site-urls` hangs and no response is sent from the controller...
I can see this ourpout:
```
$ node ./bin/www
nestedSitemapUrlObj: {
  'https://www.zara.com/robots.txt': [
    'https://www.zara.com/sitemaps/sitemap-ad-ca.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ad-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ad-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ad-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ae-ar.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ae-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-al-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-am-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-at-de.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-at-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-au-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-aw-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-az-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ba-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-be-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-be-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-be-nl.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-bg-bg.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-bg-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-bh-ar.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-bh-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-br-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-br-pt.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-by-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-by-ru.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ca-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ca-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ch-de.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ch-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ch-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ch-it.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cl-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cl-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-co-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-co-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cr-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cr-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cy-el.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cy-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cz-cs.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-cz-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-de-de.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-de-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-dk-da.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-dk-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-do-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-do-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-dz-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-dz-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ec-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ec-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ee-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ee-et.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-eg-ar.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-eg-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-es-ca.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-es-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-es-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-es-eu.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-es-gl.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-fi-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-fr-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-fr-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ge-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-gr-el.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-gr-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-gt-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-gt-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hk-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hk-zt.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hn-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hn-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hr-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hr-hr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-hu-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ic-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ic-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-id-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-id-id.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-ie-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-il-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-il-he.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-ca-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-ca-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-ca.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-en-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-en-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-en.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-es-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-es-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-es.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-fr-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-fr-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ad-fr.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-ar-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-ar-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-ar.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-en-1.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-en-2.xml.gz',
    'https://www.zara.com/sitemaps/sitemap-images-ae-en.xml.gz',
    ... 614 more items
  ]
}

but that's it
```